pkg/service/backup/restore_worker_tables.go

```diff
@@ -99,6 +100,37 @@ func (w *tablesWorker) locationRestoreHandler(ctx context.Context, run *RestoreR
 	return w.forEachRestoredManifest(ctx, w.location, manifestHandler)
 }
 
+func (w *tablesWorker) initRemainingBytesMetric(ctx context.Context, run *RestoreRun, target RestoreTarget) {
+	for _, location := range target.Location {
+		err := w.forEachRestoredManifest(
+			ctx,
+			location,
+			func(miwc ManifestInfoWithContent) error {
+				sizePerTableAndKeyspace := make(map[string]map[string]int64)
+				err := miwc.ForEachIndexIterWithError(
+					target.Keyspace,
+					func(fm FilesMeta) error {
+						if sizePerTableAndKeyspace[fm.Keyspace] == nil {
+							sizePerTableAndKeyspace[fm.Keyspace] = make(map[string]int64)
+						}
+						sizePerTableAndKeyspace[fm.Keyspace][fm.Table] += fm.Size
+						return nil
```

Comment:
> It looks like this initRemainingBytesMetric does not work well with pausing and resuming restore, since it sets remainingBytes to total table size despite the fact, that this table might be partially or fully restored in the previous run. I guess that setting remainingBytes to total table size and decreasing it in clonePrevProgress should do the trick (similar approach as in case of filesRestoredBytes).

Good finding !
But I think the solution must be different though. When the task is resumed, then the metrics shouldn't be initialized at all.
There is no option to restore this metric different way, as `remainingBytes` = a lot of smaller timeseries and each of the timeserie keeps its own value. We are not saving this information anywhere. 

Situation complicates even more if someone restart scylla-manager service between pausing and resuming. It will reset `remainingBytes` metric.... On the other hand... this is rather edge case. Besides metric will go down starting from 0 then. So... it will still keep some value.

I'll change it the way that if the task is resumed, then the metric is not reinitialized.

> Moreover, this function is really similar to newUnits, so we should combine them into one.

After quick check, I see that `newUnits` is called before knowing if the task is resumed or not.
We cannot combine it into one.
I was thinking about reusing the data from `RestoreUnit`, but it lacks the information about the source node.